src/pseudopeople/loader.py

```diff
@@ -108,9 +119,10 @@ def combine_joint_filers(data: pd.DataFrame) -> pd.DataFrame:
     # Merge spouses
     reference_persons_wide = reference_persons.merge(
         joint_filers,
-        left_on=COLUMNS.household_id.name,
-        right_on=COLUMNS.spouse_household_id.name,
+        how="left",
+        left_on=[COLUMNS.household_id.name, COLUMNS.tax_year.name],
+        right_on=[COLUMNS.spouse_household_id.name, COLUMNS.spouse_tax_year.name],
     )
-    joint_1040 = pd.concat([reference_persons_wide, independent_filers]).reset_index()
+    joint_1040 = pd.concat([reference_persons_wide, independent_filers])
```

Comment:
All of the updates here in loader are to allow for `year=None` to be passed to the noising function.